src/doomtype.h

```diff
@@ -87,13 +87,19 @@ typedef byte lighttable_t;
      __typeof__ (b) _b = (b); \
      _a > _b ? _a : _b;       \
  })
+ #define BETWEEN(l, u, x)              \
+ ({                                    \
+     __typeof__ (l) _l = (l);          \
+     __typeof__ (u) _u = (u);          \
+     __typeof__ (x) _x = (x);          \
+     _l > _x ? _l : _x > _u ? _u : _x; \
+ })
 #else
  #define MIN(a, b) (((a) < (b)) ? (a) : (b))
  #define MAX(a, b) (((a) > (b)) ? (a) : (b))
+ #define BETWEEN(l, u, x) ((l) > (x) ? (l) : (x) > (u) ? (u) : (x))
```

Comment:
I recommend a review of any use of a macro (`MIN`, `MAX`, `BETWEEN`, etc.) with calculations in the parameters.

[Example](https://github.com/fabiangreffrath/woof/blob/70aa28dc827836a0d0bba8d5cb60830af9380a40/src/r_things.c#L702):

![example](https://github.com/user-attachments/assets/338ec804-2b4b-4e91-a3ca-7f86253aaac3)

Reply:
As an alternative, we can replace these macros with `inline` functions.

Reply:
I guess it needs `auto` from C++.

Reply:
```c
inline double between(double l, double l, double x);
inline int betweeni(int l, int u, int x);
inline float betweenf(float l, float u, float x);
```
Also maybe we should rename it to `clamp`.